freqtrade/optimize/backtesting.py

```diff
@@ -1,231 +1,302 @@
+                    continue  # skip rows where no buy signal or that would immediately sell off
+
+                if realistic:
+                    if lock_pair_until is not None and row.date <= lock_pair_until:
+                        continue
+                if max_open_trades > 0:
+                    # Check if max_open_trades has already been reached for the given date
+                    if not trade_count_lock.get(row.date, 0) < max_open_trades:
+                        continue
+
+                    trade_count_lock[row.date] = trade_count_lock.get(row.date, 0) + 1
 
-    # Append Total
-    tabular_data.append([
-        'TOTAL',
-        len(results.index),
-        results.profit_percent.mean() * 100.0,
-        results.profit_BTC.sum(),
-        results.duration.mean(),
-        len(results[results.profit_BTC > 0]),
-        len(results[results.profit_BTC < 0])
-    ])
-    return tabulate(tabular_data, headers=headers, floatfmt=floatfmt)
-
-
-def get_sell_trade_entry(pair, buy_row, partial_ticker, trade_count_lock, args):
-    stake_amount = args['stake_amount']
-    max_open_trades = args.get('max_open_trades', 0)
-    trade = Trade(open_rate=buy_row.close,
-                  open_date=buy_row.date,
-                  stake_amount=stake_amount,
-                  amount=stake_amount / buy_row.open,
-                  fee=exchange.get_fee()
-                  )
-
-    # calculate win/lose forwards from buy point
-    for sell_row in partial_ticker:
-        if max_open_trades > 0:
-            # Increase trade_count_lock for every iteration
-            trade_count_lock[sell_row.date] = trade_count_lock.get(sell_row.date, 0) + 1
-
-        buy_signal = sell_row.buy
-        if should_sell(trade, sell_row.close, sell_row.date, buy_signal, sell_row.sell):
-            return sell_row, (pair,
-                              trade.calc_profit_percent(rate=sell_row.close),
-                              trade.calc_profit(rate=sell_row.close),
-                              (sell_row.date - buy_row.date).seconds // 60
-                              ), sell_row.date
-    return None
-
-
-def backtest(args) -> DataFrame:
+                ret = self._get_sell_trade_entry(pair, row, ticker[index + 1:],
+                                                 trade_count_lock, args)
+
+                if ret:
+                    row2, trade_entry, next_date = ret
+                    lock_pair_until = next_date
+                    trades.append(trade_entry)
+                    if record:
+                        # Note, need to be json.dump friendly
+                        # record a tuple of pair, current_profit_percent,
+                        # entry-date, duration
+                        records.append((pair, trade_entry[1],
+                                        row.date.strftime('%s'),
+                                        row2.date.strftime('%s'),
+                                        row.date, trade_entry[3]))
```

Comment:
is default=str  done to fix #535 (`--export trades` not working)? 
if yes, then it changes behaviour of the extract exporting `row.date` twice in different formats (once in seconds, once as timestamp).

i fixed it in #551 (I think having `row.date` there twice is an error).